p2p/src/authenticated/mod.rs

```diff
@@ -389,6 +457,60 @@ mod tests {
         for waiter in waiters {
             waiter.await.unwrap();
         }
+
+        // Shutdown last peer
+        runtime_handler.unwrap().abort();
```

Comment:
Unfortunately, this doesn't actually stop the tasks spawned by the network (it just stops the one listening for the others to end).

I'm still pondering how we should best test these sort of shutdown cases (with the deterministic runtime we can even enforce that everything actually stops at some point: https://github.com/commonwarexyz/monorepo/pull/209)

Reply:
To implement this test, we need a way to stop a running network (and others determine that the connections are no longer alive).

The tough part is that with the runtime "shutdown" handler that it destroys the entire runtime (and this test uses the same runtime).

Reply:
The "easy way out" would be to add a shutdown function just to p2p (that disconnects from all peers and exits) but that is much better handled by a stock runtime shutdown.

Reply:
There may be a way to "wrap" the shutdown trait such that it doesn't lead to entire runtime shutdown (so we can simulate this) but again, still pondering.

Reply:
If the deterministic runtime could arbitrary delay message delivery, we could end up causing something very similar to this but we've thus far implemented wrappers around the deterministic runtime elsewhere (like in p2p::simulated).

Reply:
TL;DR...this is a great test to add but I'm not in love with how we'd need to do it to get it in rn.